### DISCUSSION TITLE
Default library context scope

### DISCUSSION THREAD

**Discussion by afshinpir** [openssl/openssl#24638] [Q&A] [answered]
Hi all

What is the scope of default library context (nullptr)? Is it whole application? or it is current thread?

Best Regards

**Comment by paulidale** [comment]
It's the whole application.

**Comment by t8m** [comment, marked as answer]
However please note there are two defaults. The global default library context and a thread local default library context. Unless OSSL_LIB_CTX_set0_default() is called, these two are the same. However by calling this function you can override the default library context for the current thread.

**Comment by afshinpir** [comment]
Thanks a lot for the info.
